src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,242 @@
-# An implementation would go here!
+import subprocess
+from pathlib import Path
+from zipfile import ZipFile
+
+from requests import exceptions as requests_exceptions
+
+from briefcase.commands import (
+    BuildCommand,
+    CreateCommand,
+    PackageCommand,
+    PublishCommand,
+    RunCommand,
+    UpdateCommand,
+)
+from briefcase.config import BaseConfig
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+from briefcase.integrations.adb import (
+    DeviceNotFound, force_stop_app, install_apk, start_app)
+
+
+class ApkMixin:
+    output_format = "apk"
+    platform = "android"
+
+    @property
+    def android_sdk_path(self):
+        return Path.home() / ".briefcase" / "tools" / "android_sdk"
+
+    def binary_path(self, app):
+        return (
+            self.platform_path
+            / app.formal_name
+            / "app"
+            / "build"
+            / "outputs"
+            / "apk"
+            / "debug"
+            / "app-debug.apk"
+        )
+
+    def distribution_path(self, app):
+        return self.binary_path(app)
+
+    def verify_tools(self):
+        """
+        Verify that we can download a support package for this Python version,
+        then ensure the Android development tools are installed.
+        """
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError("""\
+Found Python version {self.python_version_tag}. Android packaging currently
+requires Python 3.7.""".format(self=self))
+
+        if not self.android_sdk_path.exists():
+            print("Setting up Android SDK...")
+            try:
+                android_sdk_zip_path = self.download_url(
+                    url=self.android_sdk_url,
+                    download_path=Path.home() / ".briefcase" / "tools",
+                )
+            except requests_exceptions.ConnectionError:
+                raise NetworkFailure("downloading Android SDK")
+            with ZipFile(android_sdk_zip_path) as android_sdk_zip:
+                android_sdk_zip.extractall(path=self.android_sdk_path)
+            # Remove the ZIP file; it has no purpose now that it is extracted.
+            android_sdk_zip_path.unlink()
+            # Set executable permissions. The ZipFile module does not extract
+            # these permissions, but Linux & macOS need them.
+            # TODO: Test this on Windows.
+            tools_bin = self.android_sdk_path_tmp / "tools" / "bin"
+            for binpath in tools_bin.glob('*'):
+                binpath.chmod(0o755)
+
+            print("Ensuring all Android SDK licenses are accepted...")
+            self.subprocess.run(
+                [tools_bin / "sdkmanager", "--licenses"],
+                check=True,
+                cwd=self.android_sdk_path,
+            )
```

Comment:
This should check for CalledProcessError.

Reply:
+1

Reply:
Fixed in a follow-up, pushed to this pull request.